student/src/main/java/tds/student/sql/repository/MessageRepository.java

```diff
@@ -108,6 +112,8 @@ public MessageRepository () {
   * If the translation is not found then will return the app key that was
   * passed in.
   */
+  @Override
+  @Cacheable(CacheType.LongTerm)
   public String getMessage (String contextType, String context, String appKey, String language) throws ReturnStatusException {
```

Comment:
Adding message caching.  I'd prefer to add it at the Service layer rather than the Repository layer, but the Service converts to MessageDTO objects into a structure with circular references...